src/renderer/base/CSSLengthPercentage.cpp

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "precomp.h"
+
+#include "../inc/CSSLengthPercentage.h"
+
+CSSLengthPercentage CSSLengthPercentage::FromString(const wchar_t* str)
+{
+    auto& errnoRef = errno; // Nonzero cost, pay it once.

```

Comment:
What is the cost and how does this ensure we pay it only once? Is it the thread safety?

Reply:
I think it's because `errno` is a thread-local variable and this gets the reference to the actual storage early. I took the code and the comment from the STL.